### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [cdump/evmole#18]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance - the same observation was mentioned in a Maturin [issue](https://github.com/PyO3/maturin/issues/1529#issue-1637803986). If you want to read a bit more about LTO, I can recommend starting from [this](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) Rustc documentation.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the package. If we enable it on the Cargo profile level, users, who build the package manually will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

According to my quick local tests with the latest `evmole` version on Fedora 41 + Rustc 1.82, adding `lto = true` to the Release profile reduces the final wheel size from 313 Kib to 284 Kib (built with the `maturin build --release` command).

Thank you.

**Comment by cdump** [comment]
Hi!

Thanks! It's been enabled in the master branch: https://github.com/cdump/evmole/commit/932284ae54da8a3c4acd121f10d34bd1515c6317

P.S. I tried using LTO some time ago but observed no changes in the benchmark results, so I didn’t enable it then. However, the reduction in size is a sufficient reason to enable it by default now.
